src/utilities/data/Matrix.hpp

```diff
@@ -33,22 +33,22 @@ UTILITIES_API bool operator!=(const Matrix& lhs, const Matrix& rhs);
 /// linear interpolation of the function v = f(x, y) at point xi, yi
 /// assumes that x and y are strictly increasing
 UTILITIES_API double interp(const Vector& x, const Vector& y, const Matrix& v, double xi, double yi, InterpMethod interpMethod = LinearInterp,
-                            ExtrapMethod extrapMethod = NoneExtrap);
+                            ExtrapMethod extrapMethod = NoneExtrap, double ti = 0.0);
```

Comment:
Please adjust the docstring of all of these to explain what the ti is, and **when** you should pass a non-zero value.